engine/src/main/java/org/terasology/persistence/internal/ReadWriteStorageManager.java

```diff
@@ -369,6 +371,14 @@ private void addGameManifestToSaveTransaction(SaveTransactionBuilder saveTransac
         for (WorldInfo worldInfo: worlds) {
             gameManifest.addWorld(worldInfo);
         }
+
+        WorldGenerator worldGenerator = CoreRegistry.get(WorldGenerator.class);
```

Comment:
wait, a GameManifest has worlds. the line right above this one just made sure of that. so why do we need to pull _another_ WorldGenerator out of CoreRegistry and add it to _different_ fields?

Reply:
because though WorldInfo has a field named `worldGenerator`, that is only the URI of the world generator class, not an initialized WorldGenerator instance we can read configuration information from.

(Unless that URI contains all the generator's configuration encoded as URL parameters, but I am pretty sure that is not the case.)

So we still need to add this world configuration information to the manifest. I think it would be good to have it integrated with the worlds field that is already there somehow. Would it be appropriate to expand the WorldInfo class to have the config information? Is there some other data class that serves that need better? I feel like there are a dozen classes with `World` in the name and I haven't sorted them out yet.

Reply:
there is an absurd amount of unecessary juggling here that is a bit maddening to look at.